susemanager-utils/testing/docker/scripts/schema_idempotency_test_pgsql.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/python
+import argparse
+import difflib
+import distutils.version as DV
+import json
+import os
+import re
+import shutil
+import subprocess
+import sys
+
+
+def file_exists(path):
+    try:
+        open(path, 'r')
+        return(True)
+    except FileNotFoundError:
+        return(False)
+
+
+def directory_exists(path):
+    if os.path.isdir(path):
+        return(True)
+    return(False)
+
+
+def get_all_files_rpm_pr(prFile):
+    with open(prFile, 'r') as f:
+        changedFiles = []
+        files = json.loads(read(f))['files']
+        for file in filesDict:
+            if file['status'] == 'modified' or file['status'] == 'added':
+                changedFiles.append(file['filename'])
```

Comment:
The currently-in-fashion way to do this kind of collection computation is by using list comprehensions (functional programming). In this case it is something like:

```python
changedFiles = [file['filename'] for file in filesDict if file['status'] == 'modified' or file['status'] == 'added']
```

Reply:
Fixed and testing.

Reply:
In this case I realized I did not add the regex to look if the file is a SQL file from the migration scripts.

So now the line looks:
```
changed_files = [file['filename'] for file in files if re.search('^schema/spacewalk/upgrade/susemanager-schema-\d+\.\d+[\d|.]*-to-susemanager-schema-\d+\.\d+[\d|.]*/.+$', file['filename']) and (file['status'] == 'modified' or file['status'] == 'added'])
```

And while I agree this in fashion, I would say it's harder to read than:
```
changed_files = [ ]
for file in files:
    if re.search('^schema/spacewalk/upgrade/susemanager-schema-\d+\.\d+[\d|.]*-to-susemanager-schema-\d+\.\d+[\d|.]*/.+$', file['filename']) and (file['status'] == 'modified' or file['status'] == 'added']):
        changed_files.append(file['filename'])
```
Thoughts?

Reply:
:+1:  for the readable one ake 2nd